src/body.rs

```diff
@@ -169,3 +169,43 @@ where
         self.body.is_end_stream()
     }
 }
+
+fn http1_headermap_to_http02(h: http_1::HeaderMap) -> http_02::HeaderMap {
+    unsafe {
+        std::mem::transmute::<http_1::HeaderMap, http_02::HeaderMap>(h)
+    }
```

Comment:
Yeah this was originally just a temporary hack. However, I am struggling to actually remove it. I pushed up a commit which kind of works, but its fairly terrible. Most notably the `extension` part of Request/Response doesn't actually seem convertable, unless I am missing some way.

The rest did work though, although the conversion is pretty terrible. Likely can use some improvements there, I don't think its robust/correct.

I pushed up another commit with these changes to iterate on

Reply:
> However, I am struggling to actually remove it.

It might compile but I really doubt it actually works. The memory layout for `Extensions` for example has changed it it now requires `Clone`. That'll most likely cause nasty UB.

> Most notably the extension part of Request/Response doesn't actually seem convertable, unless I am missing some way

I don't think there is a way to convert extensions since you cannot iterate all the values :/ I'm not sure there is much we can do about it so we have to do some testing and see how much that breaks. I'm pretty sure its gonna break web sockets because they rely on the [`OnUpgrade`](https://docs.rs/hyper/latest/hyper/upgrade/struct.OnUpgrade.html) extension which is set by hyper and would thus be removed when converting to http 0.2 extensions.

Reply:
What do you think is the path forward here?

Seems like our options are:

1. transmute - unsafe and almost certainly broken. Definitely don't like this approach
2. Convert what we can - should work for some cases but break others
3. Change `http` directly (possibly in both versions?) to expose things we need
4. Do nothing; wait for rest of ecosystem to catch up to hyper 1.0

Not sure about others but I am personally only using this for `tonic`. (2) seems viable enough for that use case (although it wasn't extensively tested yet). (4) also might be fine as well, assuming `tonic` adds hyper 1.0 support in the coming weeks/months.

Reply:
OK I took a stab at it. Not great but best I could come up with.  Can probably improve a bit, but for Extension we would need changes to `http`